src/i_winmusic.c

```diff
@@ -734,11 +835,6 @@ static void I_WIN_StopSong(void *handle)
     {
         MidiError("midiStreamStop", mmr);
     }
-    mmr = midiOutReset((HMIDIOUT)hMidiStream);
```

Comment:
For some reason, with this PR, hot swapping stopped working for me. I fixed it with this:
<details>
  <summary>diff</summary>

```diff
diff --git a/src/i_winmusic.c b/src/i_winmusic.c
index 954299c..dd72711 100644
--- a/src/i_winmusic.c
+++ b/src/i_winmusic.c
@@ -821,15 +821,16 @@ static void I_WIN_StopSong(void *handle)
 {
     MMRESULT mmr;
 
-    if (hPlayerThread)
+    if (!hPlayerThread)
     {
-        SetEvent(hExitEvent);
-        WaitForSingleObject(hPlayerThread, INFINITE);
-
-        CloseHandle(hPlayerThread);
-        hPlayerThread = NULL;
+        return;
     }
 
+    SetEvent(hExitEvent);
+    WaitForSingleObject(hPlayerThread, INFINITE);
+    CloseHandle(hPlayerThread);
+    hPlayerThread = NULL;
+
     mmr = midiStreamStop(hMidiStream);
     if (mmr != MMSYSERR_NOERROR)
     {
@@ -1008,22 +1009,32 @@ static void I_WIN_ShutdownMusic(void)
         MidiError("midiStreamRestart", mmr);
     }
     WaitForSingleObject(hBufferReturnEvent, INFINITE);
-
-    mmr = midiStreamClose(hMidiStream);
+    mmr = midiStreamStop(hMidiStream);
     if (mmr != MMSYSERR_NOERROR)
     {
-        MidiError("midiStreamClose", mmr);
+        MidiError("midiStreamStop", mmr);
     }
 
-    hMidiStream = NULL;
-
     if (buffer.data)
     {
+        mmr = midiOutUnprepareHeader((HMIDIOUT)hMidiStream, &MidiStreamHdr,
+                                     sizeof(MIDIHDR));
+        if (mmr != MMSYSERR_NOERROR)
+        {
+            MidiError("midiOutUnprepareHeader", mmr);
+        }
         free(buffer.data);
         buffer.data = NULL;
+        buffer.size = 0;
+        buffer.position = 0;
     }
-    buffer.size = 0;
-    buffer.position = 0;
+
+    mmr = midiStreamClose(hMidiStream);
+    if (mmr != MMSYSERR_NOERROR)
+    {
+        MidiError("midiStreamClose", mmr);
+    }
+    hMidiStream = NULL;
 
     CloseHandle(hBufferReturnEvent);
     CloseHandle(hExitEvent);
```
</details>